crates/polars-core/src/chunked_array/logical/categorical/builder.rs

```diff
@@ -572,6 +254,85 @@ impl CategoricalChunked {
         CategoricalChunked::from_cats_and_rev_map_unchecked(cats, Arc::new(rev_map), ordering)
     }
 
+    pub(crate) unsafe fn from_keys_and_values_global(
+        name: &str,
+        keys: impl IntoIterator<Item = Option<u32>> + Send,
+        capacity: usize,
+        values: &Utf8Array<i64>,
+        ordering: CategoricalOrdering,
+    ) -> Self {
+        // Vec<u32> where the index is local and the value is the global index
+        let mut local_to_global: Vec<u32> = Vec::with_capacity(values.len());
+        let (id, local_to_global) = crate::STRING_CACHE.apply(|cache| {
+            // locally we don't need a hashmap because we all categories are 1 integer apart
+            // so the index is local, and the values is global
+            for s in values.values_iter() {
+                // Safety: we allocated enough
+                unsafe { local_to_global.push_unchecked(cache.insert(s)) }
+            }
+            local_to_global
+        });
+
+        let compute_cats = || {
```

Comment:
Somehow this feels redundant with snippet L150 in `crates/polars-core/src/chunked_array/logical/categorical/builder.rs`.

Can we resuse that logic?